action/protocol/staking/protocol.go

```diff
@@ -798,33 +839,62 @@ func (p *Protocol) needToWriteCandsMap(ctx context.Context, height uint64) bool
 	return height >= p.config.PersistStakingPatchBlock && fCtx.CandCenterHasAlias(height)
 }
 
-func (p *Protocol) contractStakingVotes(ctx context.Context, candidate address.Address, height uint64) (*big.Int, error) {
+func (p *Protocol) contractStakingVotes(ctx context.Context, candidate address.Address, height uint64, view *ViewData) (*big.Int, error) {
 	featureCtx := protocol.MustGetFeatureCtx(ctx)
 	votes := big.NewInt(0)
 	indexers := []ContractStakingIndexer{}
+	views := []ContractStakeView{}
 	if p.contractStakingIndexer != nil && featureCtx.AddContractStakingVotes {
-		indexers = append(indexers, p.contractStakingIndexer)
+		if featureCtx.CreatePostActionStates {
+			views = append(views, view.contractsStake.v1)
+		} else {
+			indexers = append(indexers, p.contractStakingIndexer)
+		}
 	}
 	if p.contractStakingIndexerV2 != nil && !featureCtx.LimitedStakingContract {
-		indexers = append(indexers, p.contractStakingIndexerV2)
+		if featureCtx.CreatePostActionStates {
+			views = append(views, view.contractsStake.v2)
+		} else {
+			indexers = append(indexers, p.contractStakingIndexerV2)
+		}
 	}
 	if p.contractStakingIndexerV3 != nil && featureCtx.TimestampedStakingContract {
-		indexers = append(indexers, p.contractStakingIndexerV3)
+		if featureCtx.CreatePostActionStates {
+			views = append(views, view.contractsStake.v3)
+		} else {
+			indexers = append(indexers, p.contractStakingIndexerV3)
+		}
+	}
+
+	accountingVotes := func(v *VoteBucket) {
+		if v.isUnstaked() {
+			return
+		}
+		if featureCtx.FixContractStakingWeightedVotes {
+			votes.Add(votes, p.calculateVoteWeight(v, false))
+		} else {
+			votes.Add(votes, v.StakedAmount)
+		}
+	}
+	if !featureCtx.CreatePostActionStates {
```

Comment:
remove this `if`, since the for loop of `indexer.BucketsByCandidate` is always needed?